app/models/message.rb

```diff
@@ -31,6 +31,25 @@ def create_from_queue(body)
     self
   end
 
+  def set_status!
+    successes = 0
+
+    transmissions = Transmission.where(:message_uuid => self.uuid)
+    transmissions.each do |t|
+      successes += 1 if not t['state'].eql?('failed')
+    end
+
+    if successes.eql?(transmissions.length)
+      self.status = 'successful'
+    elsif successes.eql?(0)
```

Comment:
Based on above comments as it's now an array of successful transmissions you can do;

```
elsif successes.empty?
```